crypto/ecdh_extra/ecdh_test.cc

```diff
@@ -104,6 +104,17 @@ TEST(ECDHTest, TestVectors) {
     ASSERT_TRUE(EC_KEY_set_public_key(key.get(), pub_key.get()));
     ASSERT_TRUE(EC_KEY_check_key(key.get()));
 
+    // Check EVP_PKEY_check and EVP_PKEY_public_check
```

Comment:
Do we have any tests with bad keys we can add this too?

Reply:
we can add `EVP_PKEY_check` to every test where `RSA_check_key` and `EC_KEY_check_key` are  tested. that would cover "bad" keys.